fenicsadapter/config.py

```diff
@@ -0,0 +1,55 @@
+"""This is configuration module of fenicsadapter."""
+
+import os, sys, json
+
+class Config(object):
+    """Handles reading of config. parameters of the fenicsadapter based on JSON
+    configuration file. Initializer calls readJSON() method.
+
+    :ivar _config_file_name: name of the configuration file
+    :ivar _solver_name : name of the solver
+    :ivar _coupling_mesh_name: name of mesh as defined in preCICE config
+    :ivar _read_data_name: name of read data as defined in preCICE config
+    :ivar _write_data_name: name of write data as defined in preCICE config
+    :func readJSON(): method accessing data in JSON file
+    """
+
+    def __init__(self):
+        self.readJSON()
+
+        self._config_file_name
```

Comment:
There is `self.readJSON()` call before, which initializes all instance attributes (this is missing in your example). However, I can initialize them with `None` to avoid confusion, or get rid of them in `__init__`  and let the `readJSON()` create them (but it might be confusing as well).

Reply:
As `readJSON` is a quite short function, I would propose to just put all the functionality into `__init__`. Then we avoid confusion with members of the object being declared outside of the `__init__` function.